bot/helpers.py

```diff
@@ -81,12 +95,134 @@ def show_detail_menu(update, context):
+            keyboard.append([button])
+
+    try_message(context=context,
+                chat_id=query['message']['chat']['id'],
+                text=text,
+                reply_markup=InlineKeyboardMarkup(keyboard))
+
+
+def vote_on_proposal_details(update, _):
+    query = update.callback_query
+    proposal_id = query.data.split("-")[-1]
+    user_id = query.from_user['id']
+
+    try:
+        proposal = get_proposal_by_id(proposal_id)
+    except Exception as e:
+        logger.error(e)
+        query.edit_message_text(NETWORK_ERROR_MSG)
+        return
+
+    keyboard = [[InlineKeyboardButton('⬅️ BACK', callback_data='governance_active')]]
+    message = ''
+
+    if not is_wallet_provided():
+        message = f"😢 *You can't vote, no MNEMONIC key provided.* 😢"
+    elif user_id not in ALLOWED_USER_IDS:
+        message = f'😢 *You are not allowed to vote because your id ({user_id}) is not whitelisted!* 😢'
+    else:
+        try:
+            my_vote = get_my_vote(proposal_id)
+        except Exception as e:
+            logger.error(e, exc_info=True)
+            query.edit_message_text(NETWORK_ERROR_MSG)
+            return
+
+        if my_vote:
+            message = f'🎉 You voted *{my_vote}* 🎉'
+        else:
+            keyboard = [
+                [
+                    InlineKeyboardButton('✅ Yes', callback_data=f'vote-{proposal_id}-{MsgVote.YES}'),
+                    InlineKeyboardButton('❌ No', callback_data=f'vote-{proposal_id}-{MsgVote.NO}'),
+                ],
+                [
+                    InlineKeyboardButton('❌❌ No with veto',
+                                         callback_data=f'vote-{proposal_id}-{MsgVote.NO_WITH_VETO}'),
+                    InlineKeyboardButton('🤷 Abstain', callback_data=f'vote-{proposal_id}-{MsgVote.ABSTAIN}'),
+                ],
+                [InlineKeyboardButton('⬅️ BACK', callback_data='governance_active')]
+            ]
+
+    message += f"\n\n\n{jigu_proposal_to_text(proposal)}"
+
+    query.edit_message_text(message, parse_mode='markdown', reply_markup=InlineKeyboardMarkup(keyboard))
+
+
+def on_vote_clicked(update, _):
+    query = update.callback_query
+    _, proposal_id, vote = query.data.split("-")
+
+    keyboard = [[
+        InlineKeyboardButton('YES ✅', callback_data=f'vote_confirmed-{proposal_id}-{vote}'),
+        InlineKeyboardButton('NO ❌', callback_data=f'proposal-{proposal_id}')
+    ]]
+    text = f'⚠️ Do you really want to vote *{vote}* on proposal *{proposal_id}*? ⚠️'
+
+    query.edit_message_text(text, parse_mode='markdown', reply_markup=InlineKeyboardMarkup(keyboard))
+
+
+def vote_accept(update, _):
+    query = update.callback_query
+    _, proposal_id, vote = query.data.split("-")
+    keyboard = [[InlineKeyboardButton('⬅️ BACK', callback_data=f'proposal-{proposal_id}')]]
+
+    try:
+        vote_result = vote_on_proposal(proposal_id=proposal_id, vote_option=vote)
```

Comment:
When I didn't specify an invalid mnemonic and I vote, this try catch sends:

`😱 There was an error while getting data 😱
Either your internet connection or the node endpoint is down!`

Wouldn't it make more sense to return to the user that a mnemonic is provided which isn't from a validator?

Reply:
I handled the situation where the mnemonic is incorrect. 
Isn't it that everyone can vote, not only validator? In localterra I can vote as a `test3` for example but the vote just doesn't count (but it is visible).

Reply:
@mr-rooftop please don't work on this until this PR is merged because i did changes in this piece of code